src/Persistent/Implementation.hs

```diff
@@ -0,0 +1,17 @@
+{-# LANGUAGE CPP #-}
```

Comment:
I would like to avoid CPP.  Two reasons:

- complicates and destabilizes parsing, e.g. with hlint.
- makes harder to write code like e.g. "run one backend of this type and one of that; call the same actions on both; compare the result".

The benefit of CPP would be that the production code would be less bulky, but I don't want to optimize for this yet.

I think it's quite simple to replace the CPP-switch with a config-based switch.  Am I missing something?  Any other opinions?